matchtree.go

```diff
@@ -1332,6 +1332,12 @@ func pruneMatchTree(mt matchTree) (matchTree, error) {
 		mt.child, err = pruneMatchTree(mt.child)
```

Comment:
makes me think the same thing would happen for `type:file` queries then?

Reply:
good point, I will check